src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -304,68 +304,94 @@ function checkPageExistsAndRedirect(event) {
   return false;
 }
 
+/**
+ * Check if the corresponding url is absolute and make a absolute path from root if necessary
+ *
+ * @param {string} url the url to check
+ */
+async function fetchVersionSwitcherJSON(url) {
+  // first check if it's a valid URL
+  try {
+    var result = new URL(url);
+  } catch (err) {
+    // if not, assume relative path and fix accordingly
+    if (err instanceof TypeError) {
+      // workaround for redirects like https://pydata-sphinx-theme.readthedocs.io
+      // fetch() automatically follows redirects so it should work in every builder
+      // (RDT, GitHub actions, etc)
+      const origin = await fetch(window.location.origin, {
+        method: "HEAD",
+      }).then((res) => {
+        return res.url;
+      });
+      result = new URL(url, origin);
```

Comment:
it's [considered bad practice](https://stackoverflow.com/a/54497100) to mix `async/await` with `.then()`.

```suggestion
      });
      result = new URL(url, origin.url);
```